bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/AccountServlet.java

```diff
@@ -499,14 +497,8 @@ private void handleIds(HttpServletResponse resp, Connection connection, Device d
     private void renderCapabilities(Connection connection, Device device, StringBuilder html) {
         html.append("<h2>Capabilities</h2>");
         html.append("<table><tr><th align='left'>Name</th></tr>");
-        String[] capabilities = device.capabilities;
-        if (capabilities != null) {
-            for (String capability : capabilities) {
-                html.append("<tr><td>");
-                html.append(StringEscapeUtils.escapeHtml(capability));
-                html.append("</td></tr>");
-            }
-        }
+        device.getCapabilities().forEach(
+                capability -> html.append("<tr><td>" + StringEscapeUtils.escapeHtml(capability) + "</td></tr>"));
```

Comment:
You shouldn't use string concatenation when you already have a StringBuilder to use. String concatenation ends up using another StringBuilder when it generates the java bytecode. So just chain multiple `append` statements instead.
```suggestion
                capability -> html.append("<tr><td>").append(StringEscapeUtils.escapeHtml(capability)).append("</td></tr>"));
```

Reply:
Done. Also in other places.